sklearn/datasets/_openml.py

```diff
@@ -788,8 +893,32 @@ def fetch_openml(
     if as_frame == "auto":
         as_frame = not return_sparse
 
-    if as_frame and return_sparse:
-        raise ValueError("Cannot return dataframe with sparse data")
+    if parser == "auto":
+        if return_sparse:
+            parser = "liac-arff"
+        else:
+            try:
+                import pandas  # noqa
+
+                parser = "pandas"
+            except ImportError:
+                parser = "liac-arff"
+    elif parser == "pandas":
+        check_pandas_support("fetch_openml with parser='pandas'")
```

Comment:
The exception message should mention that it's possible to use `parser="liac-arff"` and `as_frame=False` to avoid the pandas dependency.